Developer Guide (#12)

blackonyyx opened the pull request:
Preliminary version.

blackonyyx committed:
Add initial and tentative documentation and plantUML diagrams to populate the Inventoryinator documentation.

This provides a base of documentation for future additions and testing

Currently some parts may still be using AddressBook3 notation but will be
iteratively removed out of the documentation.

blackonyyx committed:
Add addtional sections to Documentation to better compartementalise
portions of Inventoryinator Documentation.

blackonyyx committed:
Flesh out Usecase and segment them into feature usecases and Workflow
usecases.

Usecases are to derive from Personas in a seperate file.

To resolve stuff

blackonyyx committed:
Amend code standard to fit checkstyle requirements for CI

pass CI now

aqcd commented on docs/AboutUs.md:
```diff
@@ -5,55 +5,62 @@ title: About Us
+* Role: Developer
+* Responsibilities: Documentation, BugTesting
+
+### Liu Zhendao
+
+![Liu Zhendao](images/profiles/johndoe.png)
```
Name typo (?), img not updated

aqcd commented on docs/AboutUs.md:
```diff
@@ -5,55 +5,62 @@ title: About Us
 
-* Role: Team Lead
-* Responsibilities: UI
+![shijinglin](images/profiles/stephentan.jpg)
```
[stephentan]

aqcd commented on docs/AboutUs.md:
```diff
@@ -5,55 +5,62 @@ title: About Us
+### Kheng Hun
 
-<img src="images/johndoe.png" width="200px">
+![Kheng Hun](images/profiles/johndoe.png)
```
change names in [ ]s to be consistent styling

img not updated

aqcd commented on docs/AboutUs.md:
```diff
@@ -5,55 +5,62 @@ title: About Us
+### Rahul
 
-### James Doe
+![Rahul](images/profiles/johndoe.png)
```
img not updated

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -137,60 +137,51 @@ This section describes some noteworthy details on how certain features are imple
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+* `VersionedInventoryinator#commit()` — Saves the current address book state in its history.
+* `VersionedInventoryinator#undo()` — Restores the previous address book state from its history.
+* `VersionedInventoryinator#redo()` — Restores a previously undone address book state from its history.
```
address book -> item/recipe

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -137,60 +137,51 @@ This section describes some noteworthy details on how certain features are imple
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitInventoryinator()`, causing the modified state of the address book after the `del Bob’s 28th finger` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
 
 ![UndoRedoState1](images/UndoRedoState1.png)
 
-Step 3. The user executes `add n/David …​` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 3. The user executes `addi Bob’s 6th regret -q 8` to add a new person. The `add` command also calls `Model#commitInventoryinator()`, causing another modified address book state to be saved into the `inventoryinatorStateList`.
 
 ![UndoRedoState2](images/UndoRedoState2.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+**Note:** If a command fails its execution, it will not call `Model#commitInventoryinator()`, so the address book state will not be saved into the `inventoryinatorStateList`.
 
-</div>
-
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoInventoryinator()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
```
these steps as well

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -283,74 +254,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
```
persons -> items/recipes

aqcd commented on docs/EndUserPersona.md:
```diff
@@ -0,0 +1,150 @@
+## Persona
+### Persona #1
+#### Stephen the Hardcore Gamer (Age 21)
+![Anon](images/profiles/johndoe.png)
```
actual pictures pls :)

blackonyyx replied:
be careful what you wish for :)

aqcd commented on docs/Usecases.md:
```diff
@@ -0,0 +1,99 @@
+2a2. User inputs new data
+Steps 2a1-2a2 repeat until data is correct, or user enters a different command.
+
+3b. `inv` detects item already exists<br>
+3b1. `inv` adds quantity of item from command into registered item.
+4b. `inv` displays that item quantity is updated.
```
should this be 3a and 4a?

aqcd commented on docs/Usecases.md:
```diff
@@ -0,0 +1,99 @@
+- MSS:
+    1. User enters in input to create an recipe for an item
+    2. `inv` validates the user inputs and parses the fields given
+    3. `inv` creates and registers the recepie for an item
```
recipe

aqcd commented on docs/UserGuide.md:
```diff
@@ -14,149 +16,219 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
 1. Ensure you have Java `11` or above installed in your Computer.
 
-1. Download the latest `addressbook.jar` from [here](https://github.com/se-edu/addressbook-level3/releases).
+1. Download the latest `inventoryinator.jar` from [here](https://github.com/AY2021S1-CS2103T-F13-1/tp/releases).
```
standardise with README, use InvInator.jar or change README to inventoryinator.jar

aqcd commented on src/main/java/seedu/address/logic/commands/EditCommand.java:
```diff
@@ -167,7 +167,6 @@ public void setPhone(Phone phone) {
         public Optional<Phone> getPhone() {
             return Optional.ofNullable(phone);
         }
-
```
add line back

aqcd commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -51,7 +51,6 @@ public Email getEmail() {
     public Address getAddress() {
         return address;
     }
-
```
add line back

aqcd commented on src/main/java/seedu/address/ui/PersonCard.java:
```diff
@@ -63,12 +63,10 @@ public boolean equals(Object other) {
         if (other == this) {
             return true;
         }
-
```
add line back

aqcd commented on src/main/java/seedu/address/ui/PersonCard.java:
```diff
@@ -63,12 +63,10 @@ public boolean equals(Object other) {
         if (!(other instanceof PersonCard)) {
             return false;
         }
-
```
add line back

blackonyyx replied:
adding these lines back cause checkstyle to fail somehow?

aqcd commented on src/test/java/seedu/address/testutil/PersonBuilder.java:
```diff
@@ -88,7 +88,6 @@ public PersonBuilder withEmail(String email) {
         this.email = new Email(email);
         return this;
     }
-
```
add line back

aqcd reviewed (commented):
Preliminary review. Will do deeper checks on finalised version.

blackonyyx committed:
Add Sequence Diagrams Drawn but not used in prior iterations of Dev
Guide

This will utilise assets previously created but not utilised.

blackonyyx committed:
Add images and deletion of AB3 depreciated docs

Add new images, amend README.md to include logo, this will improve
meme-value

Add headers of documentation to all md files.

This makes it clearer as to what is happening in these Documents.

Reorganisation of assets, assest have been moved to more relevent file
directories.

Makes it easier to navigate and link images in documentation.

Deletion of AB3 Documentation. Deletion of Irrelevent and Boilerplate
Documentation from AB3 such as Tutorials.

blackonyyx committed:
Clean up of documentation and formatting of code.

blackonyyx committed:
Readd changes to codestyle to conform to checkstyle requirements to pass CI tests

justacasul commented on build/resources/main/view/PersonListCard.fxml:
```diff
@@ -31,6 +31,7 @@
       <Label fx:id="phone" styleClass="cell_small_label" text="\$phone" />
       <Label fx:id="address" styleClass="cell_small_label" text="\$address" />
       <Label fx:id="email" styleClass="cell_small_label" text="\$email" />
+      <Label fx:id="remark" styleClass="cell_small_label" text="\$remark" />
```
Do we need PersonListCard.fxml?

blackonyyx replied:
Will not touch. should be handled by #20

justacasul commented on docs/DeveloperGuide.md:
```diff
@@ -202,155 +217,72 @@ The following activity diagram summarizes what happens when a user executes a ne
 ##### Aspect: How undo & redo executes
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Saves the entire inventory.
```
What about recipes?

blackonyyx replied:
Inventory is meant to refer to entire state of the inventoryinator. will add term to glossary

justacasul commented on docs/DeveloperGuide.md:
```diff
@@ -202,155 +217,72 @@ The following activity diagram summarizes what happens when a user executes a ne
-1. _{ more test cases …​ }_
+* **Item**: An item represents an object you obtain in a game. Eg a Rock
+* **Recipe**: An recipe is associated with multiple items, and represents the consumption of items in the input,
+ to produce an item of the output.
```
Might be good to add an example like you did with **Item**

justacasul commented on docs/Testing.md:
```diff
@@ -1,13 +1,45 @@
+
+1. Saving window preferences
+
+   1. Resize the window to an optimum size. Move the window to a different location. Close the window.
```
your optimal size / an optimal size / the optimum size might be better

justacasul commented on docs/Testing.md:
```diff
@@ -1,13 +1,45 @@
+1. _{ more test cases …​ }_
 
+### Deleting a entry
+
```
Should a TODO be placed here?

justacasul commented on docs/Testing.md:
```diff
@@ -34,3 +65,11 @@ This project has three types of tests:
    e.g. `seedu.address.logic.LogicManagerTest`
+
+## UnitTesting
+
```
Might be good to have TODO here as well

justacasul commented on docs/team/liuzhendao.md:
```diff
@@ -0,0 +1,46 @@
+---
```
my zheng in the filename is missing a g lmao

blackonyyx replied:
Oops :p lmao

justacasul commented on docs/DeveloperGuide.md:
```diff
@@ -53,7 +68,7 @@ For example, the `Logic` component (see the class diagram given below) defines i
 The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
 
-<img src="images/ArchitectureSequenceDiagram.png" width="574" />
+![architecture](images/ArchitectureSequenceDiagram.png)
```
might want to update the png as well

justacasul commented on docs/DeveloperGuide.md:
```diff
@@ -78,54 +93,46 @@ The `UI` component,
 ![Structure of the Logic Component](images/LogicClassDiagram.png)
```
same here

justacasul commented on docs/DeveloperGuide.md:
```diff
@@ -137,60 +144,68 @@ This section describes some noteworthy details on how certain features are imple
+ the modified state of the inventory after the `del Bob’s 28th finger` command executes to be
+ saved in the `inventoryinatorStateList`, and the `currentStatePointer` is shifted to the newly inserted inventory state.
 
 ![UndoRedoState1](images/UndoRedoState1.png)
```
might want to update the commands

justacasul commented on docs/DeveloperGuide.md:
```diff
@@ -137,60 +144,68 @@ This section describes some noteworthy details on how certain features are imple
-Step 3. The user executes `add n/David …​` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 3. The user executes `addi Bob’s 6th regret -q 8` to add a new item. The `add` command also calls `Model#commitInventoryinator()`, causing another modified inventory state to be saved into the `inventoryinatorStateList`.
 
 ![UndoRedoState2](images/UndoRedoState2.png)
```
same here

blackonyyx replied:
Not known how implementation will be done, hence maintaining similar format as prior implementation for now.

justacasul reviewed (changes requested):
Nothing big, just change up some .png(s)

blackonyyx committed:
Add suggestions from review.

blackonyyx committed:
Add changes to profile pictures etc, for TP requirements

Add white space to Testing.md to pass CI

aqcd commented on docs/AboutUs.md:
```diff
@@ -2,58 +2,67 @@
 We are a team based in the [School of Computing, National University of Singapore](http://www.comp.nus.edu.sg).
 
-You can reach us at the email `seer[at]comp.nus.edu.sg`
+You can reach us via the issues of this repository.
```
via the issue tracker of this repository, with link to it

aqcd commented on docs/AboutUs.md:
```diff
@@ -2,58 +2,67 @@
+### Kheng Hun
 
-<img src="images/johndoe.png" width="200px">
+![khenghun](images/profiles/johndoe.png)
```
keep names consistent - full name or given name

aqcd commented on docs/CommandSequenceDiagram.md:
```diff
@@ -0,0 +1,24 @@
+
+#Command Implementation Details
+This will detail the planned sequence flow for commands, and help the developer
+understand under the hood what is going on.
```
rephrase to "what is going on under the hood"

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -51,9 +66,9 @@ For example, the `Logic` component (see the class diagram given below) defines i
 **How the architecture components interact with each other**
 
-The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `del stone`.
```
should this be "deli stone" for the moment?

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -78,54 +93,46 @@ The `UI` component,
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
-</div>
+![Interactions Inside the Logic Component for the `delete 1` Command](images/commandseqdiagrams/DeleteRecipeSequenceDiagram.png)
```
delr -n <name> -i <index>?

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -78,54 +93,46 @@ The `UI` component,
 * can save `UserPref` objects in json format and read it back.
-* can save the address book data in json format and read it back.
+* can save the inventory data in json format and read it back.
+* can save recipes data in json format and read it back. 
```
should locations be included as well?

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -137,60 +144,68 @@ This section describes some noteworthy details on how certain features are imple
 ![UndoRedoState0](images/UndoRedoState0.png)
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `del Bob’s 28th finger` command to delete the matching item in the inventory. The `del`
```
del -> deli

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -137,60 +144,68 @@ This section describes some noteworthy details on how certain features are imple
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `del Bob’s 28th finger` command to delete the matching item in the inventory. The `del`
+command calls `Model#commitInventoryinator()`, causing
+ the modified state of the inventory after the `del Bob’s 28th finger` command executes to be
```
same for this line

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -202,155 +217,73 @@ The following activity diagram summarizes what happens when a user executes a ne
 * **Alternative 2:** Individual command knows how to undo/redo by
   itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
+  * Pros: Will use less memory (e.g. for `delete`, just save the item/recipe being deleted).
```
delete -> deli/delr

aqcd commented on docs/DeveloperGuide.md:
```diff
@@ -202,155 +217,73 @@ The following activity diagram summarizes what happens when a user executes a ne
 ### \[Proposed\] Data archiving
 
-_{Explain here how the data archiving feature will be implemented}_
+TO_BE_DISCUSSED
```
mark as TODO for easier search

aqcd commented on docs/EndUserPersona.md:
```diff
@@ -0,0 +1,157 @@
+
+Through this, we come up with a set of UserStories we would like to implement as features in `Inventoryinator` 
+
+## Persona
```
optionally provide a short intro on how we structured our personas (explain that our app will help transition the before scenario to the after scenario, etc.)

aqcd commented on docs/Testing.md:
```diff
@@ -1,13 +1,46 @@
+
+1. Initial launch
+
+   1. Download the jar file and copy into an empty folder
```
include option to use

```
java -jar InvInator.java
```

aqcd commented on docs/Testing.md:
```diff
@@ -1,13 +1,46 @@
+
+   1. Download the jar file and copy into an empty folder
+
+   1. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.
```
contacts -> items

aqcd commented on docs/Testing.md:
```diff
@@ -1,13 +1,46 @@
+   1. Re-launch the app by double-clicking the jar file.<br>
+       Expected: The most recent window size and location is retained.
 
+1. _{ more test cases …​ }_
```
mark as TODO

aqcd commented on docs/Testing.md:
```diff
@@ -1,13 +1,46 @@
+
+1. Dealing with missing/corrupted data files
+
+   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+
+1. _{ more test cases …​ }_
```
mark as TODO

aqcd commented on docs/UserGuide.md:
```diff
@@ -14,165 +18,222 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
+   * **`add`** `banana -q 10` Add a quantity of 10 to the Bananas
 
-   * **`clear`** : Deletes all contacts.
+   * **`del`**`banana` : Deletes the banana item from the Inventoryinator
+
+   * **`del`**`banana` **`-r`** `1` : Deletes the first recipe from the item `bananas` 
```
should we change to deli and delr for now?

aqcd commented on docs/UserGuide.md:
```diff
@@ -14,165 +18,222 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
+**Add Recipe** | `addr` \<**product name**\> **-items** **\<item name\[quantity\]**, … > \[-pc \<num>\] \[-d \<desc\>\]
+**Add quantity to item** | `add` <**item name**> -q <**qty**>
+**List** | `list` \[-i (default) / -r\]
+**Delete entry** | `del` \<item name\> \[-r index\]
```
should we split del -> deli + delr?

aqcd commented on docs/UserGuide.md:
```diff
@@ -14,165 +18,222 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
-* `list` followed by `delete 2` deletes the 2nd person in the address book.
-* `find Betsy` followed by `delete 1` deletes the 1st person in the results of the `find` command.
+**SYNOPSIS:**
+- `del` \<item name\> \[-r index\]
```
should we split del -> deli + delr here?

aqcd commented on docs/UserGuide.md:
```diff
@@ -14,165 +18,222 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
 
 **Q**: How do I transfer my data to another Computer?<br>
-**A**: Install the app in the other computer and overwrite the empty data file it creates with the file that contains the data of your previous AddressBook home folder.
+**A**: Install the app in the other computer and overwrite the empty data file it creates with the file that contains the data of your previous Inventoryinator home folder.
```
i think we should specify that they need to overwrite multiple files, since our storage for each class is in separate files

aqcd commented on docs/team/liuzhengdao.md:
```diff
@@ -0,0 +1,46 @@
+---
+layout: page
+title: John Doe's Project Portfolio Page
```
change titles for each of these files

aqcd reviewed (commented):
Mainly need to decide if you want to change del -> deli + delr currently since it will be that way for quite a while (until v1.2 or v1.3).

blackonyyx committed:
Amend slight modifications post review.

aqcd reviewed (approved):
LGTM
